### PR TITLE
fix: remove prevent default and use react context

### LINKED ISSUE #410: Fund button is not working
**Describe the bug**

On click the button nothing happens. #361 

**To Reproduce**
1. Install the last release verdaccio `v4.4.1`
2. Load `http://localhost:4873/-/web/detail/verdaccio`
3. Click on button

**Expected behavior**
should open the targeted url

**Screenshots**

Screen Shot 2020-01-04 at 9 47 08 AM

### PR DESCRIPTION
**Pull request by juanpicado**
**Type:** bug

The following has been addressed in the PR:

*  There is a related issue? yes
*  Unit or Functional tests are included in the PR 

**Description:**

- fix the bug
- use the `context` as the other components in the same parent ( for consistence )
- [x] add a new test

Resolves #410

### CODE REVIEW

File: src/components/DetailSidebar/DetailSidebarFundButton.tsx [hunk 1]
```diff
@@ -21,21 +22,21 @@ const StyledFundStrong = styled('strong')({
...
-
 /* eslint-disable react/jsx-no-bind */
-const DetailSidebarFundButton: React.FC<Props> = ({ to }) => {
-  const preventDefault = (event: MouseEvent<HTMLButtonElement>) => event.preventDefault();
+const DetailSidebarFundButton: React.FC = () => {
+  const detailContext = useContext(DetailContext);
+
+  const { packageMeta } = detailContext;
+
+  if (!packageMeta?.latest?.funding) {
```
Comment (DanielRuf) [inline src/components/DetailSidebar/DetailSidebarFundButton.tsx]: Maybe add the `.url` check too?

File: src/components/DetailSidebar/DetailSidebarFundButton.tsx [hunk 2]
```diff
@@ -21,21 +22,21 @@ const StyledFundStrong = styled('strong')({
...
+    return null;
+  }
+
+  const to = packageMeta?.latest?.funding.url;
```
Comment (DanielRuf) [inline src/components/DetailSidebar/DetailSidebarFundButton.tsx]: Can this be merged / used with the previous line? Or reuse some variable.

File: src/components/DetailSidebar/DetailSidebarFundButton.tsx [hunk 3]
```diff
@@ -21,21 +22,21 @@ const StyledFundStrong = styled('strong')({
...
+
   return (
     <StyledLink external={true} to={to}>
-      <Button
-        color="primary"
-        fullWidth={true}
-        onClick={preventDefault}
-        startIcon={<StyledFavoriteIcon />}
-        variant="outlined">
+      <Button color="primary" fullWidth={true} startIcon={<StyledFavoriteIcon />} variant="outlined">
```
Comment (DanielRuf) [inline src/components/DetailSidebar/DetailSidebarFundButton.tsx]: Pretty long line now. Is this enforced by prettier?

Comment (juanpicado) [reply src/components/DetailSidebar/DetailSidebarFundButton.tsx]: Yes, prettier does it. :(

File: src/components/DetailSidebar/DetailSidebarFundButton.tsx (hunk 1 above)
Comment (juanpicado) [reply src/components/DetailSidebar/DetailSidebarFundButton.tsx]: According npmjs https://github.com/npm/rfcs/pull/54/files#diff-1bc199539d172d3211c84688a617e18cR30 should be there ... can we trust on them?

File: src/components/DetailSidebar/DetailSidebarFundButton.tsx (hunk 2 above)
Comment (juanpicado) [reply src/components/DetailSidebar/DetailSidebarFundButton.tsx]: I don't get it. I need to return null if funding is no there.  Am I missing something?

Comment (DanielRuf) [reply src/components/DetailSidebar/DetailSidebarFundButton.tsx]: The `to` uses the same code but with `.url` at the end. Duplicate code imo.

Also what if `.url` is not defined / set?

File: src/components/DetailSidebar/DetailSidebarFundButton.tsx (hunk 1 above)
Comment (DanielRuf) [reply src/components/DetailSidebar/DetailSidebarFundButton.tsx]: > supports a string or an object

We have to test if the metadata always contains / uses `url`.

Comment (juanpicado) [reply src/components/DetailSidebar/DetailSidebarFundButton.tsx]: I will

File: src/components/DetailSidebar/DetailSidebarFundButton.tsx [hunk 4]
```diff
@@ -21,21 +23,21 @@ const StyledFundStrong = styled('strong')({
...
-const DetailSidebarFundButton: React.FC<Props> = ({ to }) => {
-  const preventDefault = (event: MouseEvent<HTMLButtonElement>) => event.preventDefault();
+const DetailSidebarFundButton: React.FC = () => {
+  const detailContext = useContext(DetailContext);
+
+  const { packageMeta } = detailContext;
+
+  const fundingUrl = packageMeta?.latest?.funding?.url as string;
+
+  if (typeof fundingUrl !== 'string' || !isURL(fundingUrl)) {
```
Comment (DanielRuf) [inline src/components/DetailSidebar/DetailSidebarFundButton.tsx]: Hm, is the string check needed? Rest looks good.

Comment (juanpicado) [reply src/components/DetailSidebar/DetailSidebarFundButton.tsx]: True, `isUrl` does the job already.

**Comment by juanpicado** [comment]
@DanielRuf reminder :)

**Comment by DanielRuf** [comment]
Testing now.

**Pull request merged**